pinot-segment-local/src/test/java/org/apache/pinot/segment/local/utils/nativefst/mutablefst/MutableFSTConcurrentTest.java

```diff
@@ -0,0 +1,145 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.local.utils.nativefst.mutablefst;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import org.apache.commons.lang3.tuple.Pair;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.apache.pinot.segment.local.utils.nativefst.mutablefst.utils.MutableFSTUtils.regexQueryNrHitsForRealTimeFST;
+import static org.testng.AssertJUnit.assertEquals;
+
+
+public class MutableFSTConcurrentTest {
+  private ExecutorService _threadPool;
+
+  private CountDownLatch _countDownLatch;
+  private Set<String> _resultSet;
+
+  @BeforeClass
+  private void setup() {
+     _threadPool = Executors.newFixedThreadPool(2);
+    _countDownLatch = new CountDownLatch(2);
+    _resultSet = new HashSet<>();
+  }
+
+  @AfterClass
+  private void shutDown() {
+    _threadPool.shutdown();
+  }
+
+  @Test
+  public void testConcurrentWriteAndRead()
+      throws InterruptedException {
+    MutableFST mutableFST = new MutableFSTImpl();
+    List<String> words = new ArrayList<>();
+
+    words.add("ab");
+    words.add("abba");
+    words.add("aba");
+    words.add("bab");
+    words.add("cdd");
+    words.add("efg");
+
+    List<Pair<String, Integer>> wordsWithMetadata = new ArrayList<>();
+    int i = 1;
+
+    for (String currentWord : words) {
+      wordsWithMetadata.add(Pair.of(currentWord, i));
+      i++;
+    }
+
+    _threadPool.submit(() -> {
+      try {
+        performReads(mutableFST, words, 10, 200);
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+      }
+    });
+
+    _threadPool.submit(() -> {
+      try {
+        performWrites(mutableFST, wordsWithMetadata, 10);
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+      }
+    });
+
+    _countDownLatch.await();
+
+    assertEquals(_resultSet.size(), words.size());
+
+    assert (_resultSet.contains("ab"));
+    assert (_resultSet.contains("abba"));
+    assert (_resultSet.contains("aba"));
+    assert (_resultSet.contains("bab"));
+    assert (_resultSet.contains("cdd"));
+    assert (_resultSet.contains("efg"));
```

Comment:
please use `Assert.assertTrue` instead. To make it easier for the next person to work with this code, adding a message like "ab" or "\"ab\" not in the result set" would make it easier to figure out what's happened if someone breaks this code and this test catches it.